src/__internal__/focus-trap/focus-trap-utils.ts

```diff
@@ -70,7 +84,16 @@ const getNextElement = (
     if (nextElement === element) {
       // guard in case there is only one focusable element (or only a single radio group) in the trap.
       // If this happens we don't want to freeze the browser by looping forever, and it's OK to just focus
-      // the same element we're already on
+      // the same element we're already on.
+      // There is an exception though: if we're in a single radio group, we need to ensure we focus on
+      // the correct one. This may not be "element" if it's not currently focused (due to the focus actually)
```

Comment:
oops, I think that was supposed to say something more/else and I got distracted while writing that comment. I'll change it so it reflects whatever I was actually trying to say!

Reply:
looking at this now and I just put the closing `)` too early - it should be at the very end of the comment on the next line. Thanks for catching this because what I actually wrote doesn't make much sense! It's arguably still an over-long comment but I felt it was important to explain the edge case I was dealing with that made be add this check.